ffi/derive/src/convert.rs

```diff
@@ -343,3 +363,21 @@ fn derive_into_ffi_for_fieldless_enum(enum_name: &Ident, repr: &[syn::NestedMeta
         }
     }
 }
+
+fn is_opaque_wrapper(input: &DeriveInput) -> bool {
```

Comment:
Good!
I'm thinking about small trick we could do when we would implement wrapper.
Inside `ffi!` macro we can put attribute-like macro `#[opaque]` at the bottom of the item attribute list and then inside `is_opaque_wrapper` we could check presence of this attribute.
Macro `#[opaque]` will just return item as is and remove itself.

Reply:
Small [proof of concept](https://github.com/Erigara/transform).

Reply:
you seem to have a good track record of surprising me. I was considering something like this, but I thought it wouldn't work. I see no downsides to this approach. I'll name the attribute `opaque_wrapper` though. It's not enough to indicate that it's an opaque type, it's also client side.

Reply:
you seem to have a good track record of surprising me. I was considering something like this, but I thought it wouldn't work. I should name the attribute `opaque_wrapper` though. It's not enough to indicate that it's an opaque type, it's also client side.

The only downside is that `opaque_wrapper` attribute will be listed in the public API

Reply:
it can also be listed as a helper attribute in the `IntoFfi` and `TryFromReprC` derive macros

Reply:
I'll make it a derive macro attribute, I see no difference there

Reply:
> I'll make it a derive macro attribute, I see no difference there

even better this way